pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -183,24 +184,15 @@ func (a *acrProvider) Enabled() bool {
 
 func (a *acrProvider) Provide(image string) credentialprovider.DockerConfig {
 	cfg := credentialprovider.DockerConfig{}
-	ctx, cancel := getContextWithCancel()
-	defer cancel()
 
 	if a.config.UseManagedIdentityExtension {
-		klog.V(4).Infof("listing registries")
-		result, err := a.registryClient.List(ctx)
+		// Read login server from image
+		parts := strings.Split(image, "/")
```

Comment:
would you use `regexp.MustCompile` to parse login server, here is an example:
https://github.com/kubernetes/kubernetes/blob/16085784bc3899a3c343a5b2dad21ad77a45722d/pkg/volume/azure_dd/azure_common.go#L216
And also there should be error handling in case image is an invalid image url

Reply:
I have switched to use regular expression. For a public ACR, below code will handle for it
https://github.com/kubernetes/kubernetes/pull/77245/files#diff-65860720abfa0057bc9fe725fde21bedR220 . For other image registries, they will be filtered out.